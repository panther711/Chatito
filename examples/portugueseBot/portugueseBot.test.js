const fs = require("fs");
const path = require("path");
const generator = require("../../core/datasetGenerator");
const cp = require( "child_process" );

const getExampleFile = filename => path.resolve(__dirname, filename);

test("test dataset generation", () => {
    const fileContent = fs.readFileSync(getExampleFile("portugueseBot.chatito"), "utf8");
    const result = generator.datasetFromString(fileContent);
    expect(JSON.stringify(result, null, 2)).toMatchSnapshot();
});


test("test npm command line generator for snips dataset", () => {
    const d = __dirname;
    const generatedTrainingFile = path.resolve(`${d}/portugueseBot_snips_training_15.json`);
    const generatedTestingFile = path.resolve(`${d}/portugueseBot_snips_testing_21.json`);
    const npmBin = path.resolve(`${d}/../../bin/chatito.js`);
    const grammarFile = path.resolve(`${d}/portugueseBot.chatito`);
    const args = "--format=snips --max=15 --min=10";
    if (fs.existsSync(generatedTrainingFile)) { fs.unlinkSync(generatedTrainingFile); }
    if (fs.existsSync(generatedTestingFile)) { fs.unlinkSync(generatedTestingFile); }
    const child = cp.execSync(`node ${npmBin} ${grammarFile} ${args}`);
    expect(child.toString("utf8")).toMatchSnapshot();
    expect(fs.existsSync(generatedTrainingFile)).toBeTruthy();
    expect(fs.existsSync(generatedTestingFile)).toBeTruthy();
    fs.unlinkSync(generatedTrainingFile);
    fs.unlinkSync(generatedTestingFile);
    expect(fs.existsSync(generatedTrainingFile)).toBeFalsy();
    expect(fs.existsSync(generatedTestingFile)).toBeFalsy();
});

test("test npm command line generator for rasa dataset", () => {
    const d = __dirname;
    const generatedTrainingFile = path.resolve(`${d}/portugueseBot_rasa_training_15.json`);
    const generatedTestingFile = path.resolve(`${d}/portugueseBot_rasa_testing_21.json`);
    const npmBin = path.resolve(`${d}/../../bin/chatito.js`);
    const grammarFile = path.resolve(`${d}/portugueseBot.chatito`);
    const args = "--format=rasa --max=15 --min=10";
    if (fs.existsSync(generatedTrainingFile)) { fs.unlinkSync(generatedTrainingFile); }
    if (fs.existsSync(generatedTestingFile)) { fs.unlinkSync(generatedTestingFile); }
    const child = cp.execSync(`node ${npmBin} ${grammarFile} ${args}`);
    expect(child.toString("utf8")).toMatchSnapshot();
    expect(fs.existsSync(generatedTrainingFile)).toBeTruthy();
    expect(fs.existsSync(generatedTestingFile)).toBeTruthy();
    fs.unlinkSync(generatedTrainingFile);
    fs.unlinkSync(generatedTestingFile);
    expect(fs.existsSync(generatedTrainingFile)).toBeFalsy();
    expect(fs.existsSync(generatedTestingFile)).toBeFalsy();
});