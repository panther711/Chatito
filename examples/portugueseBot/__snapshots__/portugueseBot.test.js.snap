// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test dataset generation 1`] = `
"{
  \\"rasa_nlu_data\\": {
    \\"regex_features\\": [],
    \\"entity_synonyms\\": [
      {
        \\"value\\": \\"janeiro\\",
        \\"synonyms\\": [
          \\"Janeiro\\",
          \\"janero\\",
          \\"jan\\",
          \\"01\\"
        ]
      },
      {
        \\"value\\": \\"fevereiro\\",
        \\"synonyms\\": [
          \\"feverero\\",
          \\"fev\\",
          \\"02\\"
        ]
      }
    ],
    \\"common_examples\\": [
      {
        \\"text\\": \\"boleto janeiro\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 6,
            \\"value\\": \\"boleto\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 7,
            \\"end\\": 14,
            \\"value\\": \\"janeiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"boleto Janeiro\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 6,
            \\"value\\": \\"boleto\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 7,
            \\"end\\": 14,
            \\"value\\": \\"janeiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"boleto janero\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 6,
            \\"value\\": \\"boleto\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 7,
            \\"end\\": 13,
            \\"value\\": \\"janeiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"boleto jan\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 6,
            \\"value\\": \\"boleto\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 7,
            \\"end\\": 10,
            \\"value\\": \\"janeiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"boleto 01\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 6,
            \\"value\\": \\"boleto\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 7,
            \\"end\\": 9,
            \\"value\\": \\"janeiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"boleto fevereiro\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 6,
            \\"value\\": \\"boleto\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 7,
            \\"end\\": 16,
            \\"value\\": \\"fevereiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"boleto feverero\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 6,
            \\"value\\": \\"boleto\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 7,
            \\"end\\": 15,
            \\"value\\": \\"fevereiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"boleto fev\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 6,
            \\"value\\": \\"boleto\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 7,
            \\"end\\": 10,
            \\"value\\": \\"fevereiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"boleto 02\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 6,
            \\"value\\": \\"boleto\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 7,
            \\"end\\": 9,
            \\"value\\": \\"fevereiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"DAS janeiro\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 3,
            \\"value\\": \\"DAS\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 4,
            \\"end\\": 11,
            \\"value\\": \\"janeiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"DAS Janeiro\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 3,
            \\"value\\": \\"DAS\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 4,
            \\"end\\": 11,
            \\"value\\": \\"janeiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"DAS janero\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 3,
            \\"value\\": \\"DAS\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 4,
            \\"end\\": 10,
            \\"value\\": \\"janeiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"DAS jan\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 3,
            \\"value\\": \\"DAS\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 4,
            \\"end\\": 7,
            \\"value\\": \\"janeiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"DAS 01\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 3,
            \\"value\\": \\"DAS\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 4,
            \\"end\\": 6,
            \\"value\\": \\"janeiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"DAS fevereiro\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 3,
            \\"value\\": \\"DAS\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 4,
            \\"end\\": 13,
            \\"value\\": \\"fevereiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"DAS feverero\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 3,
            \\"value\\": \\"DAS\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 4,
            \\"end\\": 12,
            \\"value\\": \\"fevereiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"DAS fev\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 3,
            \\"value\\": \\"DAS\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 4,
            \\"end\\": 7,
            \\"value\\": \\"fevereiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"DAS 02\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 3,
            \\"value\\": \\"DAS\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 4,
            \\"end\\": 6,
            \\"value\\": \\"fevereiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"contribuicao janeiro\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 12,
            \\"value\\": \\"contribuicao\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 13,
            \\"end\\": 20,
            \\"value\\": \\"janeiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"contribuicao Janeiro\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 12,
            \\"value\\": \\"contribuicao\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 13,
            \\"end\\": 20,
            \\"value\\": \\"janeiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"contribuicao janero\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 12,
            \\"value\\": \\"contribuicao\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 13,
            \\"end\\": 19,
            \\"value\\": \\"janeiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"contribuicao jan\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 12,
            \\"value\\": \\"contribuicao\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 13,
            \\"end\\": 16,
            \\"value\\": \\"janeiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"contribuicao 01\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 12,
            \\"value\\": \\"contribuicao\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 13,
            \\"end\\": 15,
            \\"value\\": \\"janeiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"contribuicao fevereiro\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 12,
            \\"value\\": \\"contribuicao\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 13,
            \\"end\\": 22,
            \\"value\\": \\"fevereiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"contribuicao feverero\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 12,
            \\"value\\": \\"contribuicao\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 13,
            \\"end\\": 21,
            \\"value\\": \\"fevereiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"contribuicao fev\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 12,
            \\"value\\": \\"contribuicao\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 13,
            \\"end\\": 16,
            \\"value\\": \\"fevereiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"contribuicao 02\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 12,
            \\"value\\": \\"contribuicao\\",
            \\"entity\\": \\"boleto\\"
          },
          {
            \\"start\\": 13,
            \\"end\\": 15,
            \\"value\\": \\"fevereiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"janeiro\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 7,
            \\"value\\": \\"janeiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"Janeiro\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 7,
            \\"value\\": \\"janeiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"janero\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 6,
            \\"value\\": \\"janeiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"jan\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 3,
            \\"value\\": \\"janeiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"01\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 2,
            \\"value\\": \\"janeiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"fevereiro\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 9,
            \\"value\\": \\"fevereiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"feverero\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 8,
            \\"value\\": \\"fevereiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"fev\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 3,
            \\"value\\": \\"fevereiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      },
      {
        \\"text\\": \\"02\\",
        \\"intent\\": \\"findBoletoDASByMes\\",
        \\"entities\\": [
          {
            \\"start\\": 0,
            \\"end\\": 2,
            \\"value\\": \\"fevereiro\\",
            \\"entity\\": \\"mes\\"
          }
        ]
      }
    ]
  }
}"
`;

exports[`test npm command line generator for rasa dataset 1`] = `
"/Users/rodrigo/Trabajo/Chatito/examples/portugueseBot/portugueseBot_rasa_training_15.json
/Users/rodrigo/Trabajo/Chatito/examples/portugueseBot/portugueseBot_rasa_testing_21.json
{ total: 36, training: 15, testing: 21 }
"
`;

exports[`test npm command line generator for snips dataset 1`] = `
"/Users/rodrigo/Trabajo/Chatito/examples/portugueseBot/portugueseBot_snips_training_15.json
/Users/rodrigo/Trabajo/Chatito/examples/portugueseBot/portugueseBot_snips_testing_21.json
{ total: 36, training: 15, testing: 21 }
"
`;
